lib/types/string.js

```diff
@@ -665,6 +665,10 @@ module.exports = Any.extend({
                     return helpers.error('string.uri');
                 }
 
+                if (helpers.prefs.convert && options.encodeUri) {
+                    value = encodeURI(value);
```

Comment:
Didn't want to reassign the value from parameter because this is not a good practice, but reassigning the value to another variable might cause additional changes in the code below with `value`. Please lmk how you think, guys.

Reply:
It should be fine, we never manipulate arguments in the codebase